core/src/main/java/com/valtech/aemsaas/core/services/impl/BaseHttpServiceImpl.java

```diff
@@ -0,0 +1,191 @@
+        return executePost(url, postParameters, headers, defaultCredentialsProvider);
+    }
+
+    public BaseResponse executePost(String url, List<NameValuePair> postParameters, SearchHeaders headers, CredentialsProvider credentialsProvider) throws IOException {
+        CloseableHttpClient client = createHttpClient(credentialsProvider);
+        CloseableHttpResponse response = null;
+        HttpPost httpPost = new HttpPost(url);
+        if (headers != null) {
+            LOGGER.debug("Adding Headers to search");
+            headers.createHeaders(httpPost);
+        }
+        try {
+            LOGGER.debug("Adding post parameters to search : {}", postParameters.stream().map(NameValuePair::toString).collect(Collectors.joining(",")));
+            httpPost.setEntity(new UrlEncodedFormEntity(postParameters, Charsets.UTF_8.name()));
+            LOGGER.info("Executing POST to Search : {}", url);
+            response = client.execute(httpPost);
+            String responseString = parseResponse(response);
+            LOGGER.debug("{} : {}", url, responseString);
+            return new BaseResponse(response.getStatusLine().getStatusCode(), responseString);
+        } finally {
+            if (response != null) {
+                response.close();
+            }
+            client.close();
+        }
+    }
+
+    private CloseableHttpClient createHttpClient(CredentialsProvider credentialsProvider) {
+        httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+        httpClientBuilder.setDefaultRequestConfig(requestConfig);
+        return httpClientBuilder.build();
+    }
+
+    public SearchHeaders generateSearchHeader() {
+        return new SearchHeaders();
+    }
+
+    private String parseResponse(HttpResponse response) {
+        try (BufferedReader rd = new BufferedReader(
+                new InputStreamReader(response.getEntity().getContent(), Charsets.UTF_8.name()))) {
+
+            StringBuilder result = new StringBuilder();
+            String line;
+            while ((line = rd.readLine()) != null) {
+                result.append(line);
+            }
+
+            return result.toString();
+        } catch (IOException e) {
+            return StringUtils.EMPTY;
+        }
+    }
+
+    private HttpClientBuilder setIgnoredSSLHttpClientBuilder(HttpClientBuilder httpClientBuilder) throws KeyManagementException, NoSuchAlgorithmException, KeyStoreException {
+        SSLContextBuilder builder = new SSLContextBuilder();
+        builder.loadTrustMaterial(null, new TrustStrategy() {
+            public boolean isTrusted(X509Certificate[] chain, String authType)
+                    throws CertificateException {
+                return true;
+            }
+        });
+        SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(
+                builder.build());
+        return httpClientBuilder.setSSLSocketFactory(sslsf);
```

Comment:
Please log a warning that this shouldn't be used on production when this is enabled. Intention of this setting is to have it for self signed certificates. 
Unfortunately it is used as well where corporate Forward Proxy's interecepted SSL.